plugins/modules/nd_setup.py

```diff
@@ -0,0 +1,355 @@
+    default: present
+    choices: [ present, query ]
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Setup ND
+  cisco.nd.nd_setup:
+    cluster_name: cluster1
+    ntp_server:
+      - 179.37.212.207
+    dns_server:
+      -  210.69.111.224
+    dns_search_domain:
+      - cisco.com
+    app_network: 192.18.0.1/16
+    service_network: 200.90.0.0/16
+    nodes:
+      - hostname: Test
+        serial_number: 3C0A86H4D02E
+        management_ip_address: 13.34.56.23
+        username: rescue-user
+        password: test
+        management_network:
+          ipv4_address: 13.34.56.23/24
+          ipv4_gateway: 13.34.56.1
+        data_network:
+          ipv4_address: 12.34.56.22/24
+          ipv4_gateway: 12.34.56.1
+        bgp:
+          asn: 2
+          peers:
+            - ip: 15.36.56.22/24
+              asn: 2
+
+- name: Check the installation status of ND
+  cisco.nd.nd_setup:
+    state: query
+"""
+
+RETURN = r"""
+"""
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+from ansible_collections.cisco.nd.plugins.module_utils.nd_argument_specs import management_network_spec, data_network_spec, bgp_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        cluster_name=dict(type="str"),
+        ntp_server=dict(type="list", aliases=["ntp_servers"], elements="str"),
+        dns_server=dict(type="list", aliases=["dns_servers"], elements="str"),
+        proxy_server=dict(type="str"),
+        proxy_username=dict(type="str"),
+        proxy_password=dict(type="str", no_log=True),
+        ignore_proxy=dict(type="list", elements="str"),
+        dns_search_domain=dict(type="list", aliases=["dns_search_domains"], elements="str"),
+        app_network=dict(type="str"),
+        service_network=dict(type="str"),
+        nodes=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                hostname=dict(type="str", required=True),
+                serial_number=dict(type="str", required=True),
+                management_ip_address=dict(type="str", required=True),
+                username=dict(type="str", required=True),
+                password=dict(type="str", required=True, no_log=True),
+                management_network=dict(type="dict", required=True, options=management_network_spec()),
+                data_network=dict(type="dict", required=True, options=data_network_spec()),
+                bgp=dict(type="dict", options=bgp_spec()),
+            ),
+        ),
+        state=dict(type="str", default="present", choices=["present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["cluster_name", "ntp_server", "dns_server", "app_network", "service_network", "nodes"]],
```

Comment:
Are all these options required for setup or can we also configure without certain options?

Reply:
These are all required in the UI. Besides I tried to configure the cluster without a couple of options present above and the API threw an error. I didn't test all the options though.